lib/ansible/galaxy/api.py

```diff
@@ -781,7 +799,10 @@ def get_collection_versions(self, namespace, name):
             api_path = self.available_api_versions['v2']
             pagination_path = ['next']
 
-        versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions', '/')
+        if 'v3' in self.available_api_versions:
+            versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions', '/?limit=%s' % COLLECTION_PAGE_SIZE)
+        else:
+            versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions', '/?page_size=%s' % COLLECTION_PAGE_SIZE)
```

Comment:
I think that copy-pasting the invocation can make this prone to human error during refactoring because one would have to update two code branches w/o making typos. Mind making just the query arg name conditional?
```suggestion
        page_size_name = 'limit' if 'v3' in self.available_api_versions else 'page_size'
        versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions', '/?%s=%s' % (page_size_name, COLLECTION_PAGE_SIZE))
```